pkg/controllers/manifestwork.go

```diff
@@ -243,7 +269,7 @@ func (r *HypershiftDeploymentReconciler) deleteManifestworkWaitCleanUp(ctx conte
 
 	if m.GetDeletionTimestamp().IsZero() {
 		patch := client.MergeFrom(m.DeepCopy())
-		setManifestWorkSelectivelyDeleteOption(m, helper.GetHostingNamespace(hyd))
+		setManifestWorkSelectivelyDeleteOption(m, helper.GetHostingNamespace(hyd), hyd.Name, hyd.Spec.Override)
```

Comment:
Maybe you should just pass the `hyd` pointer instead of three of its parameters.